### PR TITLE
Batsheva/update image

### PR DESCRIPTION
**Pull request by BatshevaTavger**

### CODE REVIEW

**Comment by EA171** [comment]
please add a description to the PR - which tasks did you solve in this PR, also ask for code review from someone from your team, then send it again to me and Sari and we'll review the code.

File: client-side/src/components/settinPage.js [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+            console.error('Error uploading file:', error);
+            if(error.response)
+             console.log('Response data :',error.response.data);
+          }
+           
+    };
+    
+    const handleFilePicture = (e) => {
+      if (e) {
+        console.log('at handleFilePicture the  file is ' ,e.target.files[0]);
```
Comment (moriya2004) [inline client-side/src/components/settinPage.js:120]: delete the console.log

File: client-side/src/components/settinPage.js [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+          { audioSrc &&
+            <audio controls>
+               <source src={audioSrc} ></source>
+            </audio>}
+          </div>
+       {/*upload image section   */}
+       <div>
+       <h2>Change Profile Picture</h2>
+        <input type="file" accept="image/*" onChange={handleFilePicture} />
+        <GenericButton size='small' label='Upload Image' onClick={handleUploadPicture} className='' />
```
Comment (moriya2004) [inline client-side/src/components/settinPage.js:175]: Choose a more meaningful className

File: client-side/src/components/settinPage.js [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+        {preview &&   <div>            <img src={preview} alt="Profile Preview" style={{ width: '150px', height: '150px', objectFit: 'cover' }} />
+        </div>  }
+          
+       </div>
+            <GenericButton size='small'  label='send preference' onClick={handleUpload} className=''/>
```
Comment (moriya2004) [inline client-side/src/components/settinPage.js:180]: Choose a more meaningful className

File: client-side/src/components/settings/manageNotifications/EmailFrequency.jsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+//         size='large'
+//         widthOfSelect='210px'
+//       />
+//       /* TO DO: replace message */
+//       <p>{message}</p>
+//     </div>
+//   );
+// };
+
+// export default EmailFrequency;
```
Comment (moriya2004) [inline client-side/src/components/settings/manageNotifications/EmailFrequency.jsx:68]: remove this component from your project

File: client-side/src/components/settings/manageNotifications/TimeZone.jsx [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+//                 onChange={handleTimeZoneChange}
+//                 size='large'
+//                 widthOfSelect='200px'
+//             />
+//             <p>Current time in {timeZone}: {formatTime(new Date())}</p>
+//         </div>
+//     );
+// };
+
+// export default TimeZone;
```
Comment (moriya2004) [inline client-side/src/components/settings/manageNotifications/TimeZone.jsx:76]: delete this component from your project

File: client-side/src/components/settings/editUserProfile/ProfileImageEdit.jsx [hunk 6]
```diff
@@ -0,0 +1,58 @@
+import React, { useState } from 'react';
+import axios from 'axios';
+import GenericButton from '../../../stories/Button/GenericButton';
+
+const ProfileImageEditButton = () => {
+  const [file, setFile] = useState(null);
+  const [preview, setPreview] = useState(null);
+  const userId = '6694d2295d41f7809588274c';
```
Comment (moriya2004) [inline client-side/src/components/settings/editUserProfile/ProfileImageEdit.jsx:8]: get user from props

File: client-side/src/components/settings/editUserProfile/ProfileImageEdit.jsx [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+  const [imageFile,setImageFile]= useState(null);
+  const url=process.env.REACT_APP_BASE_URL;
+
+  const handleFilePicture = (e) => {
+    if (e) {
+      console.log('at handleFilePicture the  file is ' ,e.target.files[0]);
```
Comment (moriya2004) [inline client-side/src/components/settings/editUserProfile/ProfileImageEdit.jsx:14]: delete console.log

File: client-side/src/components/settings/editUserProfile/ProfileImageEdit.jsx [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+         reader.readAsDataURL(e.target.files[0]);
+      }
+    }
+  };
+
+  const handleUploadPicture = async () => {
+    const formData = new FormData();
+    formData.append('profileImage', imageFile);
+    try {
+      const response = await axios.put(`http://localhost:5000/users/${userId}`, formData, {
```
Comment (moriya2004) [inline client-side/src/components/settings/editUserProfile/ProfileImageEdit.jsx:31]: use ${url}
not http://localhost:5000
for exemple:`${baseUrl}/preferences/${preferenceId}`

File: client-side/src/components/settings/editUserProfile/ProfileImageEdit.jsx [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+        headers: {
+          'Content-Type': 'multipart/form-data',
+        },
+      });
+    } catch (error) {
+      if(error.response)
+      alert('Failed to update profile image.');
```
Comment (moriya2004) [inline client-side/src/components/settings/editUserProfile/ProfileImageEdit.jsx:38]: do not use alert

File: client-side/src/components/settings/editUserProfile/ProfileImageEdit.jsx [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+    }
+  };
+
+  return (
+    <div style={{ display: 'flex', flexDirection: 'column', alignItems: 'center', justifyContent: 'center' }}>
+      <div style={{ display: 'flex', flexDirection: 'row' }}>
+        <h2>Change Profile Picture</h2>
+        <input type="file" accept="image/*" onChange={handleFilePicture} />
+        <GenericButton size='small' label='Upload Image' onClick={handleUploadPicture} className='' />
```
Comment (moriya2004) [inline client-side/src/components/settings/editUserProfile/ProfileImageEdit.jsx:47]: Choose a more meaningful className

File: client-side/src/components/settings/editUserProfile/ProfileImageEdit.jsx [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+      if(error.response)
+      alert('Failed to update profile image.');
+    }
+  };
+
+  return (
+    <div style={{ display: 'flex', flexDirection: 'column', alignItems: 'center', justifyContent: 'center' }}>
+      <div style={{ display: 'flex', flexDirection: 'row' }}>
+        <h2>Change Profile Picture</h2>
+        <input type="file" accept="image/*" onChange={handleFilePicture} />
```
Comment (moriya2004) [inline client-side/src/components/settings/editUserProfile/ProfileImageEdit.jsx:46]: use the input that Shira created

**Pull request closed without merging**